sos/plugins/sapnw.py

```diff
@@ -32,13 +32,13 @@ def collect_list_instances(self):
             "/usr/sap/hostctrl/exe/saphostctrl -function ListInstances",
             suggest_filename="SAPInstances"
         )
-        if not inst_out:
+        if inst_out['status'] != 0:
             return
 
         sidsunique = set()
         # Cycle through all the instances, get 'sid', 'instance_number'
         # and 'vhost' to determine the proper profile
-        p = open(inst_out, "r").read().splitlines()
+        p = open(inst_out['filename'], "r").read().splitlines()
```

Comment:
Since this is using `collect_cmd_output`, we can just use the `output` key instead of opening the file. E.G.

`for line in inst_out['output'].splitlines():`